src/App.jsx

```diff
@@ -14,33 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getGoodsPrepared(goods, goodsSortBy, reverse) {
+  let goodsPrepared = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (goodsSortBy) {
+    goodsPrepared = goodsPrepared.sort((good1, good2) => {
+      switch (true) {
+        case goodsSortBy === 'abc':
+          return good1 > good2 ? 1 : -1;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case goodsSortBy === 'length':
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse % 2 === 0) {
+    goodsPrepared = goodsPrepared.reverse();
+  }
+
+  return goodsPrepared;
+}
+
+let COUNTER_PLUS = 1;
+
+export const App = () => {
+  const [query, setQuery] = useState('');
+  const [reverse, setReverse] = useState(1);
+  const renderingGoods = getGoodsPrepared(goodsFromServer, query, reverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': query === '' || query === 'length',
+          })}
+          onClick={() => {
+            setQuery('abc');
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': query === '' || query === 'abc',
+          })}
+          onClick={() => {
+            setQuery('length');
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': reverse % 2 !== 0,
+          })}
+          onClick={() => {
+            setReverse((COUNTER_PLUS += 1));
+          }}
+        >
+          Reverse
+        </button>
+
+        {(query || reverse % 2 === 0) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setReverse(1);
+              setQuery('');
+            }}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+      <ul>
+        {renderingGoods.map(good => (
+          <li data-cy="Good">{good}</li>
+        ))}
```

Comment:
don't forget to add the key prop
```suggestion
        {renderingGoods.map(good => (
          <li data-cy="Good" key={good} >{good}</li>
        ))}
```